Source/WebCore/rendering/mac/RenderThemeMac.mm

```diff
@@ -1015,9 +1015,16 @@ static void setFontFromControlSize(RenderStyle& style, NSControlSize controlSize
     return sizes;
 }
 
-void RenderThemeMac::adjustMenuListStyle(RenderStyle& style, const Element* e) const
+void RenderThemeMac::adjustMenuListStyle(RenderStyle& style, const Element* element) const
 {
-    RenderThemeCocoa::adjustMenuListStyle(style, e);
+#if ENABLE(VECTOR_BASED_CONTROLS_ON_MAC)
+    if (element && element->document().settings().vectorBasedControlsOnMacEnabled()) {
+        RenderThemeCocoa::adjustMenuListStyle(style, element);
+        return;
+    }
+#else
+    UNUSED_PARAM(element);
+#endif
     NSControlSize controlSize = controlSizeForFont(style);
```

Comment:
To address the other comment, I think we just need a call to `RenderTheme::adjustMenuListStyle` before this line..